client/data/settings/index.js

```diff
@@ -0,0 +1,12 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import reducer from './reducer';
+import * as selectors from './selectors';
+import * as actions from './actions';
+import * as resolvers from './resolvers';
+
+export { reducer, selectors, actions, resolvers };
```

Comment:
I have the same question here: I would export only the hooks. The other parts can always be imported directly in the `store.js` file. This way we simplify the API and make its usage clearer.
But I know this is how the other parts of the data store have been developed. This is just an open question.

Reply:
Here we have to export all the stuff because [it's registered with the store here.](https://github.com/Automattic/woocommerce-payments/pull/1647/files#diff-5c16b7bc520682552479e6ce09776baf4c03532ec9e16f3cf07d26782ec36950R16)

Reply:
Oh, that's true! I just gave the code a look though and I see that `client/data/settings/index.js` isn't actually imported anywhere aside from in `client/data/store.js` for the purpose of registering it with the store so the file only serves the purpose of encapsulating the settings importing/exporting logic in one place.

Based on that, I'd say it's OK to leave it as it is but if you feel differently, let me know!

Reply:
Yeah, that was exactly my thought! Since we're not using higher level components and we're accessing the data layer through hooks, I thought it made no sense to export all the things in `data/settings`